google/resource_compute_instance_template.go

```diff
@@ -221,6 +242,33 @@ func resourceComputeInstanceTemplate() *schema.Resource {
 										Computed: true,
 										Optional: true,
 									},
+
+									"assigned_nat_ip": &schema.Schema{
```

Comment:
Since an IP is only assigned when an instance is created, it doesn't make sense to have this field in the instance template.

Reply:
I can't think of a good way to remove this from instance templates while also sharing interface flatten code with instances. I think that might be what sent me down the path of deprecating `assigned_nat_ip` in favour of just using `nat_ip` for both explicit and computed NAT IPs.

Reply:
Part of me prefers your suggested way, but using it would require passing all of `d` down to `flattenAccessConfigs`, and I feel a little better about that function currently simply acting on a slice of *compute.AccessConfig (at the expense of having to keep a pointless `assigned_nat_ip` in the compute instance schema). I've added a comment.